### DISCUSSION TITLE
Questions about using "hideAttribution" in commercial projects?

### DISCUSSION THREAD

**Discussion by cxqeric** [xyflow/xyflow#2961] [Q&A] [answered]
Thanks to the author for developing it, I really like this library and I have seen many sites using it.
I have a question for you. If I plan to use it commercially, but I only use the basic features and not the react-flow Pro features, can I use the "Remove Attribution"?
I'm a little confused because this library is MIT protocol, however, the "free" card in this URL (https://pro.reactflow.dev/pricing) says "Best for individuals and non-commercial projects. and non-commercial projects.".
and the api doc show that:
![image](https://user-images.githubusercontent.com/26587077/227710484-83826ea0-bc0d-44ae-9191-39194c9b212c.png)
Or do I need to donate through Github? I can donate, but may be I don't have enough money to subscribe.

I don't even have any action yet because it's just an idea, and  I don't know if it will be profitable.
I saw some discussions inside discussions (https://github.com/wbkd/react-flow/discussions/2015, https://github.com/wbkd/react-flow/discussions/2520,,, https://reactflow.dev/blog/asking-for-money-for-open-source/,   https://reactflow.dev/docs/api/react-flow-props/#prooptions), but I read them with the help of translation software and am not very sure of the final conclusion.

Thanks again and I look forward to hearing from you.
I'm sorry if I've wasted your time

PS: I am not a native English speaker, the above is from translation software.

**Comment by bcakmakoglu** [comment, marked as answer]
First off there are no pro features.
All features of React Flow are available for free, none are hidden behind a paywall.
The pro subscription will give you access to the code of special examples and depending on your subscription, you have other benefits like a call with one of the developers.

But from what I understand you're not obligated to pay, although it would be greatly appreciated, especially if the library is used commercially.

**Reply by moklick** [reply]
I couldn't say it better ;) Thanks @bcakmakoglu 🙏

**Reply by santhosh-c1** [reply]
Wanted to follow up here. We love react flow, but confusion point is:

The `hideAttribute` is behind a parameter named `proOptions`. [Remove attribution doc](https://reactflow.dev/docs/guides/remove-attribution/). The comment in the code clearly states this:

```
/**
 * This example demonstrates how you can remove the attribution from the React Flow renderer.
 * Please only hide the attribution if you are subscribed to React Flow Pro: https://pro.reactflow.dev ** <---- This part **
 */
```

The MIT licence, this answer, wording in the docs and the one in the code example are very conflicting.

Can you please provide guidance on this specific question: Can we remove attribution in a commercial for-profit product, without paying/sponsoring/subscribing to react flow pro? 

Thank you!

**Reply by bcakmakoglu** [reply]
You *are allowed* to hide the attribution when using it in any application, whether commercial or not.
It is just suggested that you subscribe to React Flow Pro so that the project can be maintained in the future. Without any funds it's gonna be hard to keep working on React Flow 😄

**Reply by santhosh-c1** [reply]
Thanks for the lightning fast response! Much appreciated!

> Please only hide the attribution if you are subscribed to React Flow Pro

This wording here in https://reactflow.dev/docs/guides/remove-attribution/ just didn't feel like a suggestion IMHO. 

Having asked this question, I will totally discuss means to support you all internally.
